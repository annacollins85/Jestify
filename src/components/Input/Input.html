<div class="form-group">
  <label :for="id">{{ label }}</label>
  <input type="text" class="form-control" :name="name" v-moddel="value" :id="id" :disabled="disabled" :maxlength="maxlength" autocomplete="off"/>
  <div class="invalid-feedback"></div>
</div>
