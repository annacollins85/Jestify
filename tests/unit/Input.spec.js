import { shallow } from '@vue/test-utils'
// import { createRenderer } from 'vue-server-renderer'

import Input from '@/components/Input/Input.vue'

describe('Input.vue', () => {
  it('renders props.label when passed', () => {
    const label = 'Input label'
    const wrapper = shallow(Input, {
      propsData: { label }
    })
    expect(wrapper.text()).toMatch(label)
  })
  // it('has a max length when passed', () => {
  //   const maxLength = 10
  //   const wrapper = shallow(Input, {
  //     propsData: { maxLength }
  //   })
  //   // check tohtml has attribute maxLength
  //   console.log((wrapper.find('form-control').attributes()))
  //   expect(wrapper.props().maxLength).toBe(10)
  // })
  // it('matches snapshot', () => {
  //   const label = 'Input label'
  //   const renderer = createRenderer()
  //   const wrapper = shallow(Input, {
  //     propsData: { label }
  //   })
  //   renderer.renderToString(wrapper.vm, (err, str) => {
  //     if (err) throw new Error(err)
  //     expect(str).toMatchSnapshot()
  //   })
  // })
})
