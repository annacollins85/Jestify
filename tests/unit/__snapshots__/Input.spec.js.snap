// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input.vue matches snapshot 1`] = `
<div class="form-group">
    <label>Input label</label>
    <input type="text" autocomplete="off" class="form-control">
    <div class="invalid-feedback"></div>
</div>
`;
